lib/elements/satin_column.py

```diff
@@ -132,6 +174,32 @@ def zigzag_spacing(self):
         # peak-to-peak distance between zigzags
         return max(self.get_float_param("zigzag_spacing_mm", 0.4), 0.01)
 
+    @property
+    @param('random_zigzag_spacing',
+           _('Zig-zag spacing randomness(peak-to-peak)'),
+           tooltip=_('percentage  of randomness  of Peak-to-peak distance between zig-zags.'),
+          type='int', unit="%")
+    def random_zigzag_spacing(self):
+        # peak-to-peak distance between zigzags
+         return max(self.get_int_param("random_zigzag_spacing", 0), 0)
+
+
+
+    @property
+    @param(
+        'pull_compensation_unit',
+        _('Pull compensation_unit'),
+        tooltip=_('can be either mm or percentage'),
+        type='dropdown',
```

Comment:
It would be cleaner to add a `pull_compensation_percent` field. Pull compensation in mm and in percent compensate for different properties of the fabric and it might be useful to have both.

Reply:
I'll change that.